### DISCUSSION TITLE
Use same name for query and mutation

### DISCUSSION THREAD

**Discussion by nikitastryuk** [trpc/trpc#2276] [Ideas]
Hello! Is it possible to use same name for both query and mutation? (I'm using trpc v10).
Webhook i use sends /GET request to callback url first to check if server is up and only then sends /POST requests - so their api path has to be exactly the same.
Thanks in advance!

<img width="496" alt="image" src="https://user-images.githubusercontent.com/32720808/180221961-86602c15-0843-4ec5-ae18-681218d3afb5.png">

**Comment by jlalmes** [comment]
This is currently not supported but we definitely could (and should?) support this because it was possible in v9. 

I imagine the API could look something link this:

```ts
const appRouter = t.router({
  webhook: [
    t.procedure.query(/* ... */),
    t.procedure.mutation(/* ... */),
    t.procedure.subscription(/* ... */),
  ]
})

```

**Reply by nikitastryuk** [reply]
this looks great and simple actually

**Comment by KATT** [comment]
We might re-add support for this. For now, we won't allow it because they are saved in the same procedure record.

We might also allow method overriding (#1939) which could solve this & also allow ending a procedure by making it a sort of `.queryOrMutation`.

**Comment by KATT** [comment]
How come you're using trpc for webhook rather than a custom api handler? 🤔

I feel like you won't be able to take advantage of the input validation and stuff anyway?

**Reply by nikitastryuk** [reply]
Huge thanks for answering!
Custom handler should be the way to go!
